sgv2-service-common/src/main/java/io/stargate/sgv2/common/cql/builder/ConcreteValue.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright DataStax, Inc. and/or The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.sgv2.common.cql.builder;
+
+class ConcreteValue<T> implements Value<T> {
```

Comment:
should we also take advantage of `Immutable` here, so we get proper `hashCode` and `equals` that we are missing at the moment?

Reply:
We could, but I don't see a strong incentive given how simple the class is.
Note that we don't need value equality (in fact this type is never compared, used as map key, etc), so the default `equals` and `hashCode` are appropriate.

Reply:
I am always afraid that one day somebody will decide to use such a class instance in a map.. That's why I am always looking to ensure that such POJO classes are properly implemented.. But I get your point..